cmd/migration-assist/commands/postgres.go

```diff
@@ -108,9 +111,35 @@ func runTargetCheckCmdF(cmd *cobra.Command, args []string) error {
 		return nil
 	}
 
+	mysqlMigrations := "mysql.output"
+	if len(args) > 1 {
+		mysqlMigrations = args[1]
+	}
+
+	var src sources.Source
+
 	// download required migrations if necessary
 	migrationDir, _ := cmd.Flags().GetString("migrations-dir")
-	if migrationDir == "" {
+	if _, err = os.Stat(mysqlMigrations); !os.IsNotExist(err) {
```

Comment:
This entire if-else logic is getting a bit large to be readable. Also, we are ignoring any error from `os.Stat`. Can we refactor this entire thing to a separate function that returns the `source`? That way you don't need to declare `var src source.Source` and then assign it inside the condition. This would also remove the need to have else-ifs since you can just return the src directly from inside each case.


```
_, err := os.Stat(mysqlMigrations)
if err != nil && os.IsNotExist(err) {
  return nil, err
} else {
  // do logic, return src, nil
}

if migrationDir == "" { 
  // do logic, return src
}
// do logic, return src
```

And then outside:
```
src, err := getMigrationSource(//stuff to pass)
```

Reply:
Yep makes sense. It just needs to cover a bunch of flags and combination of those. I'll try my best.